docs/user_guide/fairness_in_machine_learning.rst

```diff
@@ -141,6 +141,132 @@ which groups these values were observed, as well as the difference and ratio
 between the maximum and the minimum values. For more information refer to the
 subpackage :code:`fairlearn.metrics`.
 
+What traps can we fall into when abstracting a social problem?
+--------------------------------------------------------------
+
+Machine learning systems used in the real world are inherently sociotechnical
+systems, which include both technologies and social actors. Designers of machine
+learning systems who fail to consider the way in which social contexts and technologies
+are interrelated are at risk of falling into "abstraction traps". In this section,
+we explain what those traps are, and give some suggestions on how you can avoid them.
+
+In "Fairness and Abstraction in Sociotechnical Systems" Selbst et al. [#4]_
+identify failure modes that arise from applying abstraction. They identify
+them as:
+
+* *The Solutionism Trap*
+
+* *The Ripple Effect Trap*
+
+* *The Formalism Trap*
+
+* *The Portability Trap*
+
+* *The Framing Trap*
+
+Each of these traps are the result of failing to consider the way in
+with social context and technology are interrelated, as well as a deeper understanding of
+"the social" in order to solve problems Ackerman [#5]_.
```

Comment:
I also wonder whether we want to include (some paraphrased, de-jargoned version of) the high-level summary of them from the takeaways section of the paper? Maybe here? Not sure - what do you think @lgfunderburk @hildeweerts 

[copy-pasted from the paper, but we'd want a less jargon-y version, probably]
(1) is appropriate to the situation in the first place, which requires a nuanced understanding of the relevant social context and its politics (Solutionism);
(2) affects the social context in a predictable way such that the problem that the technology solves remains unchanged after its introduction (Ripple Effect);
(3) can appropriately handle robust understandings of social requirements such as fairness, including the need for procedurality, contextuality, and contestability (Formalism);
(4) has appropriately modeled the social and technical requirements of the actual context in which it will be deployed (Portability); and
(5) is heterogeneously framed so as to include the data and social actors relevant to the localized question of fairness (Framing)

Reply:
Great idea @mmadaio!

Reply:
Removed original sentences from paper, and added less jargon-y versions.

Reply:
Added sentence you provided

`We provide some definitions and examples of these traps to help Fairlearn users think about how choices they make in their work can avoid these common pitfalls.`